hack/update-codegen.sh

```diff
@@ -133,9 +137,10 @@ export -f resources_groups
 operator_groups() {
   echo "Generating API groups for pkg/apis/operator"
 
-  bash "${PROJECT_ROOT}"/hack/generate-groups.sh \
+  bash "${CODE_GEN_DIR}"/generate-internal-groups.sh \
```

Comment:
On using `generate-group.sh`, I am getting `permission denied` error
```
go/pkg/mod/k8s.io/code-generator@v0.28.3/generate-groups.sh: line 66: go/pkg/mod/k8s.io/code-generator@v0.28.3/generate-internal-groups.sh: Permission denied
```
because `generate-groups.sh` is using `exec` call to call the `generate-internal-groups.sh`
https://github.com/kubernetes/kubernetes/blob/4909dbd0d7fd1ca13dc8c8db3af3cbf4733f3035/staging/src/k8s.io/code-generator/generate-groups.sh#L65-L66

I would suggest lets proceed with the current change as we will migrate to new `kube-codegen` script from the next release of kubernetes i.e `1.30`